ClimsoftVer4/ClimsoftVer4/ucrNavigation.vb

```diff
@@ -1,4 +1,112 @@
 ﻿Public Class ucrNavigation
+    Private bFirstLoad As Boolean = True
+    Public maxRows As Integer
+    Public currRow As Integer
+
+    Public Overrides Sub PopulateControl()
+        MyBase.PopulateControl()
+        currRow = 0
+        maxRows = dtbRecords.Rows.Count
+        displayRecordNumber()
+        If maxRows > 0 Then
+
+        Else
+
+        End If
+
+    End Sub
+
+    'Not sure whether it's necessary to override this
+    'and if it's to be overriden what value should it return and in what format
+
+    'Public Overrides Function GetValue() As Object
+    '    Return Nothing
+    'End Function
+
+    Public Overrides Function GetValue(strFieldName As String) As Object
+
+        If strFieldName = "" Then
+            Return GetValue()
+        End If
+
+        If dtbRecords.Rows.Count > 0 Then
+            Return dtbRecords.Rows(currRow).Field(Of Object)(strFieldName)
+        Else
+            Return ""
+        End If
+
+    End Function
+
+    Private Sub displayRecordNumber()
+        'Display the record number in the data navigation Textbox
+        If maxRows = 0 Then
+            txtRecNum.Text = "Record " & 0 & " of " & 0
+        Else
+            txtRecNum.Text = "Record " & currRow + 1 & " of " & maxRows
+        End If
+
+    End Sub
+
+    Protected Overridable Sub ucrNavigation_Load(sender As Object, e As EventArgs) Handles Me.Load
```

Comment:
In here we need to make the textbox for the navigation to be non editable- Actually maybe on an InitialiseControl sub